docs/cli/ilab-processes.md

```diff
@@ -0,0 +1,65 @@
+# Processes in InstructLab
+
+The ability to detach from processes is crucial to the user experience of InstructLab. However, the concept of multi-processing, process management, and the monitoring of processes is very complex.
+
+It is important to try and add this concept in as simply as possible, expanding on the state reporting, logging, and other features as we go along.
+
+## Phased approach to InstructLab Processes
+
+This document is going to describe phase 1 of implementing processes in InstructLab. Phase 1 is to be described as the "ilab simple process management system". This will depend purely on python packages, PID tracking, and log files to create the experience of detachable processes. The key here is the concept of the UUID, allowing a future REST API to keep track of InstructLab processes using these unique identifiers.
+
+We can re-visit all this in phase 2, when we discuss if we want to utilize something like systemd or a more in-depth process-monitor repo to track processes.
+
+### Phase 1
+
+Phase one would focus on adding the concept of detaching from processes. This would be implemented in `ilab 0.23.0`.
+
+Process management would only apply to `ilab data generate` and `ilab model train`.
+
+The workflow would allow for:
+
+`ilab data generate -dt` (run a detached generation process)
+`ilab model train -dt` (run a detached training process)
+
+`ilab process list`
+
+```console=
++----------------------------------------+------------+-------------+-------------+---------+
+| Proccess UUID                          | PID        |  Type       | Duration    | Status  |
++----------------------------------------+--------------------------+-------------+---------+
+| train-2423813fhthfbnaj                 |  33451     |  training   | 8 hr 32 min | Running |
+| generation-6768769599sdbjklm           |  35543     |  generation | 3 hr 15 min | Running |
+| generation-1345661hlsmma               |  45424     |  generation | 22 min      | Errored |
++----------------------------------------+---------------------+------------------+---------+
+
+```
```

Comment:
sorry, I didn't test before, just curious, is the host can do or there is a ability to handle multi-tasks at the same time.

Reply:
yep, you can have multiple processes at once, especially SDG if you set gpu_layers properly in your config